source/texture2darray.ts

```diff
@@ -0,0 +1,376 @@
+
+        if (internalFormat === this._internalFormat
+            && (format === undefined || format === this._format)
+            && (type === undefined || type === this._type)) {
+            return;
+        }
+        assert(internalFormat !== undefined, `valid internal format expected`);
+        this._internalFormat = internalFormat;
+
+        if (format) {
+            this._format = format;
+        }
+        if (type) {
+            this._type = type;
+        }
+
+        this.data(undefined, bind, unbind);
+    }
+
+    /**
+     * This should be used to implement efficient resize the texture.
+     * @param width - Targeted/new width of the texture in px.
+     * @param height - Targeted/new height of the texture in px.
+     * @param depth - Targeted/new depth of the texture in px.
+     * @param bind - Allows to skip binding the texture (e.g., when binding is handled outside).
+     * @param unbind - Allows to skip unbinding the texture (e.g., when binding is handled outside).
+     */
+    @Initializable.assert_initialized()
+    resize(width: GLsizei, height: GLsizei, depth: GLsizei, bind: boolean = true, unbind: boolean = true): void {
+        if (width === this._width && height === this._height && depth === this._depth) {
+            return;
+        }
+        this._width = width;
+        this._height = height;
+        this._depth = depth;
+
+        this.data(undefined, bind, unbind);
+    }
+
+    /**
+     * Returns the number of bytes this object approximately allocates on the GPU. The size will be zero when no
+     * image data was passed to the texture object.
+     */
+    get bytes(): GLsizei {
+        this.assertInitialized();
+        return this.context.allocationRegister.allocated(this._identifier);
+    }
+
+    /**
+     * Cached internal format of the texture for efficient resize. This can only be changed by re-initialization.
+     */
+    get internalFormat(): GLenum {
+        this.assertInitialized();
+        return this._internalFormat!;
+    }
+
+    /**
+     * Cached format of the data provided to the texture object for efficient resize. This is set on initialization and
+     * might change on data transfers.
+     */
+    get format(): GLenum {
+        this.assertInitialized();
+        return this._format!;
+    }
+
+    /**
+     * Cached type of the data provided to the texture used for efficient resize. This is set on initialization and
+     * might change on data transfers.
+     */
+    get type(): GLenum {
+        this.assertInitialized();
+        return this._type!;
+    }
+
+    /**
+     * The width of the texture object in px.
+     */
+    get width(): GLsizei {
+        this.assertInitialized();
+        return this._width;
+    }
+
+    /**
+     * The height of the texture object in px.
+     */
+    get height(): GLsizei {
+        this.assertInitialized();
+        return this._height;
+    }
+
+    /**
+     * The depth of the texture object in px.
```

Comment:
Somewhere else you wrote that the depth are the slices. I think this fact should belong here, too, e.g.
`The depth (number of slices) of the texture object`.
Is it still in px?